dedupe/labeler.py

```diff
@@ -142,13 +142,11 @@ def learn_predicates(
     def _predict(self, pairs: TrainingExamples) -> Labels:
         labels: Labels = []
         for record_1, record_2 in pairs:
-
             for predicate in self.current_predicates:
                 keys = predicate(record_2, target=True)
-                if keys:
-                    if set(predicate(record_1)) & set(keys):
-                        labels.append(1)
-                        break
+                if not keys.isdisjoint(predicate(record_1)):
```

Comment:
this looks great. 

we need to wrap this in a try except block, because third-party variables may not be returning a frozenset.

something like
```python
            try:
                if not keys.isdisjoint(predicate(record_1)):
                   labels.append(1)
                   break
             except AttributeError:
                warn('the predicate {} is not returning a frozen set, this will soon be required behaviour'.format(predicate.__name__), DeprecationWarning)
                if not set(keys).isdisjoint(predicate(record_1)):
                   labels.append(1)
                   break
```

Reply:
we'll need to do this everywhere we expect predicates to return a set.

Reply:
Done everywhere.